core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/definitions/ldap/CircuitBreakerLdapServiceDecorator.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ *    This file is part of ReadonlyREST.
+ *
+ *    ReadonlyREST is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    ReadonlyREST is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
+ */
+package tech.beshu.ror.accesscontrol.blocks.definitions.ldap
+
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.numeric.Positive
+import monix.eval.Task
+import tech.beshu.ror.accesscontrol.domain
+import tech.beshu.ror.accesscontrol.domain.User
+import monix.catnap._
+import tech.beshu.ror.utils.uniquelist.UniqueList
+
+import scala.concurrent.duration.FiniteDuration
+
+class CircuitBreakerLdapAuthenticationServiceDecorator(underlying: LdapAuthenticationService,
+                                                       maxFailures: Int,
```

Comment:
can we gather all the CB settings into eg `CircuitBreakerConfig` class?

_(We can do it in a second part of the task)_